app/scripts/lib/crypto/recovery-keys.js

```diff
@@ -0,0 +1,164 @@
+      });
+  },
+
+  /**
+   * Get the current version of the recovery keys.
+   *
+   * @returns {string} current recovery key version
+   */
+  getCurrentRecoveryKeyVersion: function () {
+    return getRecoveryKeyVersion();
+  },
+
+  /**
+   * Retrieves the recovery JWK from the uid and recovery key. The
+   * recoveryJwk can be used to encrypt and decrypt data.
+   *
+   * @param {String} uid Uid of user
+   * @param {String} recoveryKey Recovery key
+   * @returns {Promise} A promise that will be fulfilled with JWK
+   */
+  getRecoveryJwk: function (uid, recoveryKey) {
+    return importFxaCryptoDeriver().then(({jose}) => {
+      required(uid, 'uid');
+      required(recoveryKey, 'recoveryKey');
+
+      return Base32.decode(recoveryKey)
+        .then((keyMaterial) => {
+          keyMaterial = Buffer.from(keyMaterial, 'utf8');
+          const salt = Buffer.from(uid, 'hex');
+          const options = {
+            info: Buffer.from('fxa recovery encrypt key', 'utf8'),
+            length: 32,
+            salt
+          };
+          const optionsKid = {
+            info: Buffer.from('fxa recovery fingerprint', 'utf8'),
+            length: 16,
+            salt
+          };
+          return Promise.all([jose.JWA.derive(HKDF_SHA_256, keyMaterial, options), jose.JWA.derive(HKDF_SHA_256, keyMaterial, optionsKid)])
+            .then((result) => {
+              const recoveryEncryption = result[0].toString('hex');
+              const recoveryKeyId = result[1].toString('hex');
+              const keyOptions = {
+                alg: ENCRYPTION_ALGORITHM,
+                k: jose.util.base64url.encode(recoveryEncryption, 'hex'),
+                kid: recoveryKeyId,
+                kty: 'oct'
+              };
+              return jose.JWK.asKey(keyOptions);
+            });
+        });
+    });
+  },
+
+  /**
+   * Creates an encrypted bundle for the users's recovery data. This data
+   * should contain the user's original kB.
+   *
+   * @param {String} recoveryJwk Recovery JWK used to encrypt data
+   * @param {Object} recoveryData Data to encrypt
+   * @param {Object} [options={}] Options
+   *   @param {String} [options.iv] - Input vector used to create bundle
+   * @returns {Promise} A promise that will be fulfilled with the encrypted recoveryData
+   */
+  bundleRecoveryData: function (recoveryJwk, recoveryData, options = {}) {
+    return importFxaCryptoDeriver().then(({jose}) => {
+      required(recoveryJwk, 'recoveryJwk');
+
+      const recipient = {
+        header: {
+          alg: 'dir',
+          enc: ENCRYPTION_ALGORITHM
+        },
+        key: recoveryJwk
+      };
+      const encryptOptions = {
+        contentAlg: ENCRYPTION_ALGORITHM,
+        format: 'compact'
+      };
+
+      if (options.iv) {
+        encryptOptions.iv = jose.util.base64url.encode(options.iv, 'hex');
+      }
+
+      return jose.JWE.createEncrypt(encryptOptions, recipient)
+        .update(JSON.stringify(recoveryData))
```

Comment:
do you need to worry about `undefined` recoveryData because below you won't be able to `JSON.parse(undefined)`